File: backend/app/models/summary.py
"""Summary model for AI-generated unit summaries."""
from sqlalchemy import Column, Integer, String, Text, JSON, DateTime, func, UniqueConstraint
from app.database import Base


class Summary(Base):
    __tablename__ = "summaries"

    id = Column(Integer, primary_key=True, index=True)
    key = Column(String(255), nullable=True, unique=True, index=True)
    unit_ids = Column(JSON, nullable=False, default=list)
    content = Column(Text, nullable=False)
    key_points = Column(JSON, nullable=False, default=list)
    created_at = Column(DateTime, server_default=func.now())

    __table_args__ = (
        UniqueConstraint("key", name="uq_summary_key"),
    )

File: frontend/src/services/api/types/summary.ts
// SyllabusAI summary types.
// Extracted from services/api.ts (F8 split).

// ----- SyllabusAI summaries -----
export interface Summary {
  id: number;
  unit_ids: number[];
  content: string;
  key_points: string[];
  created_at?: string | null;
}

export interface SummaryGenerateResponse {
  summary: { content: string; key_points: string[] };
  cached: boolean;
}
